src/model/keychain.ts

```diff
@@ -193,13 +221,38 @@ export async function saveObject(
     logger.DebugContext.keychain
   );
   const jsonValue = JSON.stringify(value);
+
+  // only save public data to mmkv
+  // private data has accessControl
+  if (!accessControlOptions?.accessControl) {
+    saveStringMMKV(key, jsonValue);
+  }
+  // save to keychain
   return saveString(key, jsonValue, accessControlOptions);
 }
 
 export async function loadObject(
   key: string,
   options?: Options
 ): Promise<null | Record<string, any> | -1 | -2> {
+  // try to load data from mmkv first
+  // if its private or does not exist, it will return undefined
+  const jsonValueMMKV = loadStringMMKV(key);
+
+  if (jsonValueMMKV) {
+    try {
+      const objectValue = JSON.parse(jsonValueMMKV);
+      return objectValue;
+    } catch (err) {
+      logger.error(
+        new RainbowError(
+          `Keychain MMKV: failed to parse object for key: ${key} error: ${err}`
```

Comment:
Should kill this `key` prop since some keys contain PII. I'd put the stringified error into the second arg to `logger.error` too so that the value of the `RainbowError` is consistent in Sentry and are bucketed together.